app/Http/Controllers/User/ReviewBookController.php

```diff
@@ -61,25 +61,29 @@ public function store(ReviewRequest $request, $slug)
             $owners = $this->book->find($id, ['owners'])->owners;
             $request->merge(['book_id' => $id]);
 
-            $review = $this->review->store($request->all());
-            if ($review && $owners) {
-                foreach ($owners as $owner) {
-                    $info = [
-                        'send_id' => Auth::id(),
-                        'receive_id' => $owner->id,
-                        'target_type' => config('model.target_type.review'),
-                        'target_id' => $review->id,
-                        'viewed' => config('model.viewed.false'),
-                    ];
-                    $this->notification->store($info);
+            $isReview = $this->review->checkReview($request->all())->toArray();
+            if(empty($isReview)) {
+                $review = $this->review->store($request->all());
+                if ($review && $owners) {
+                    foreach ($owners as $owner) {
+                        $info = [
+                            'send_id' => Auth::id(),
+                            'receive_id' => $owner->id,
+                            'target_type' => config('model.target_type.review'),
+                            'target_id' => $review->id,
+                            'viewed' => config('model.viewed.false'),
+                        ];
+                        $this->notification->store($info);
+                    }
                 }
+                $data = $this->review->find($id);
+                if (isset($data) && $data != null) {
+                    $this->book->updateStar($data, $id);
+                }
+                return redirect()->route('books.show', $slug);
             }
-            $data = $this->review->find($id);
-            if (isset($data) && $data != null) {
-                $this->book->updateStar($data, $id);
-            }
+            return back();
 
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingBeforeClose</br>**Severity**: error</br>**File**: app/Http/Controllers/User/ReviewBookController.php</br>Blank line found at end of control structure